Allow "ubuntu-latest" runners for all workflows (#1146)
microsoft/AL-Go: RELEASENOTES.md

In the summary after a Test Run, you now also have the result of performance tests.

### Ubuntu runners for all AL Go workflows

Previously, the workflows "Update AL Go System Files" and "Pull Request Handler" were excluded when changing the two parameters to prevent deadlocks and security issues.

> From now on, a check will only allow changing the parameters of the blacklisted workflows if the `runs-on` parameter is either `windows-latest` or `ubuntu-latest`. Additionally, a check in the function `ModifyRunsOnAndShell` ensures only the value `pwsh` is allowed when using an `ubuntu-latest` runner.

Reviewer: ```suggestion
From now on, a check will only allow changing the parameters of the excluded workflows if the `runs-on` parameter is either `windows-latest` or `ubuntu-latest`. Additionally, only value `pwsh` for `shell` setting is allowed when using an `ubuntu-latest` runner (`runs-on` setting).
```
